### PR TITLE
Better handle non-git repositories

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring

## :scroll: Description
Disable preferences and options based on availability of git to completely avoid the class of bugs pertaining to non-git repositories.

## :bulb: Motivation and Context
There have been multiple requests over the year for a 'proper' no-Git mode and this gives us a start in that direction.

## :green_heart: How did you test it?
Start by having a git repository

## :pencil: Checklist

- [x] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code

### CODE REVIEW

**Comment by Skrilltrax** [comment]
@msfjarvis can we make the PasswordRepository a singleton, I think having multiple objects of a repository might cause issues further down the line.

**Comment by msfjarvis** [comment]
> @msfjarvis can we make the PasswordRepository a singleton, I think having multiple objects of a repository might cause issues further down the line.

We need to at some point, but I also wish to keep both these types separate so maybe the solution for this is going to be a static field initialized on init. Maybe Dagger needs to make an appearance after all :smirk:

File: app/src/main/java/com/zeapo/pwdstore/repository/GitRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    override fun getRepository(): Repository? = repository
+
+    override fun getFilesList(path: File): ArrayList<File> {
+        if (!path.exists()) return arrayListOf()
+        return path.listFiles { file -> file.isDirectory || file.extension == ".gpg" }
+                ?.toCollection(ArrayList()) ?: arrayListOf()
+    }
+
+    override fun listPasswords(path: File, rootDir: File, sortOrder: PasswordSortOrder): ArrayList<PasswordItem> {
```
Comment (FabianHenneke) [inline app/src/main/java/com/zeapo/pwdstore/repository/GitRepositoryImpl.kt:49]: Would it be worth it to extract the similar logic in SearchableRepositoryViewModel and reuse it here?

Comment (msfjarvis) [reply app/src/main/java/com/zeapo/pwdstore/repository/GitRepositoryImpl.kt:49]: Probably yeah. I'll look into it once the no-git implementation is done.

**Comment by Skrilltrax** [comment]
> > @msfjarvis can we make the PasswordRepository a singleton, I think having multiple objects of a repository might cause issues further down the line.
> 
> We need to at some point, but I also wish to keep both these types separate so maybe the solution for this is going to be a static field initialized on Application init. Maybe Dagger needs to make an appearance after all 😏

I'm not really a fan of separate types. A user may create a repository on phone and then add git afterwards. Can't we something like isGitRepo to check if the repository is a git repository before performing git functions ?

**Comment by msfjarvis** [comment]
> > > @msfjarvis can we make the PasswordRepository a singleton, I think having multiple objects of a repository might cause issues further down the line.
> > 
> > We need to at some point, but I also wish to keep both these types separate so maybe the solution for this is going to be a static field initialized on Application init. Maybe Dagger needs to make an appearance after all 😏
> 
> I'm not really a fan of separate types. A user may create a repository on phone and then add git afterwards. Can't we something like isGitRepo to check if the repository is a git repository before performing git functions ?

Hm that's a fair enough point... We already have the checks, but many git functions are spread around the source when they should be concentrated in the PasswordRepository class. I'll drop the existing stuff here and just reuse the PR to do that instead.

**Comment by msfjarvis** [comment]
I removed the interface based reimplementation and decided to just skip things based on availability of git.

**Pull request merged**